packages/replayio/src/commands/record.ts

```diff
@@ -9,44 +9,79 @@ import { printRecordings } from "../utils/recordings/printRecordings";
 import { selectRecordings } from "../utils/recordings/selectRecordings";
 import { LocalRecording } from "../utils/recordings/types";
 import { uploadRecordings } from "../utils/recordings/upload/uploadRecordings";
+import { logPromise } from "../utils/async/logPromise";
 
 registerCommand("record", { checkForRuntimeUpdate: true, requireAuthentication: true })
   .argument("[url]", `URL to open (default: "about:blank")`)
   .description("Launch the replay browser in recording mode")
   .action(record);
 
 async function record(url: string = "about:blank") {
-  const recordingsBefore = await getRecordings();
+  const prevRecordings = await getRecordings();
 
   await launchBrowser(url, { processGroupId: uuid() });
 
   const recordingsAfter = await getRecordings();
-  const recordingsNew = recordingsAfter.filter(
-    recording => !recordingsBefore.some(({ id }) => id === recording.id)
-  );
+
+  const nextCrashedRecordings: LocalRecording[] = [];
+  const nextRecordings: LocalRecording[] = [];
+
+  recordingsAfter.filter(recording => {
+    if (!prevRecordings.some(({ id }) => id === recording.id)) {
+      if (recording.recordingStatus === "crashed") {
+        nextCrashedRecordings.push(recording);
+      } else {
+        nextRecordings.push(recording);
+      }
+    }
+  });
 
   console.log(""); // Spacing for readability
 
-  if (recordingsNew.length > 0) {
+  // First check for any new crashes; these we should upload automatically
+  if (nextCrashedRecordings.length > 0) {
+    console.log(
+      "It looks like something went wrong with this recording. Please hold while we upload crash data."
+    );
+
+    const promise = uploadRecordings(nextCrashedRecordings, {
+      processAfterUpload: false,
+      silent: true,
+    });
+
+    logPromise(promise, {
+      messages: {
+        pending: "Uploading crash data...",
+        success: "Crash data uploaded successfully",
+      },
+    });
+
+    await promise;
```

Comment:
if this fails - do we want to halt everything?

Reply:
> Other types of errors would get caught by createSettledDeferred and resolved with a status of false (rather than throwing).

Right. So that means that it can't ever fail here. And since it's configured with `silent: true` we end up with a very broken situation in which we report `✔ Crash data uploaded successfully` when the upload of the crashed data has failed.

Reply:
> Right. So that means that it can't ever fail here. And since it's configured with silent: true we end up with a very broken situation in which we report ✔ Crash data uploaded successfully when the upload of the crashed data has failed.

Hmm. I tested the failure case for this PR before posting it, and we didn't incorrectly report upload success.

I'm having a bit of trouble keeping up with bouncing back and forth between PRs though. If you think this is broken, can you post a PR with a proposed change and/or demonstrate it failing? (I believe you. It just seems more efficient that way than chatting on an already merged PR.)